src/System.IO.Compression/tests/CompressionStreamUnitTests.Gzip.cs

```diff
@@ -32,6 +32,137 @@ public void Precancellation()
+            using (var decompressorOutput = new MemoryStream())
+            {
+                for (int i = 0; i < streamCount; i++)
+                {
+                    using (var gz = new GZipStream(compressedStream, CompressionLevel.NoCompression, true))
+                    {
+                        for (int j = 0; j < bytesPerStream; j++)
+                        {
+                            byte b = (byte)((i * j) % 256);
+                            gz.WriteByte( b);
+                            correctDecompressedOutput.WriteByte(b);
+                        }
+                    }
+                }
+                compressedStream.Seek(0, SeekOrigin.Begin);
+
+                var decompressor = CreateStream(compressedStream, CompressionMode.Decompress);
+                //Thread.Sleep(10000);
+
+                var bytes = new byte[bufferSize];
+                bool finished = false;
+                int retCount = 0, totalRead = 0;
+                while (!finished)
+                {
+                    if (useAsync)
+                    {
+                        try
+                        {
+                            retCount = await decompressor.ReadAsync(bytes, 0, bufferSize);
+                            totalRead += retCount;
+                            if (retCount != 0)
+                                await decompressorOutput.WriteAsync(bytes, 0, retCount);
+                            else
+                                finished = true;
+                        }
+                        catch (Exception)
+                        {
+                            throw new Exception(retCount + " " + totalRead);
+                        }
+                    }
+                    else if (useReadByte)
+                    {
+                        int b = decompressor.ReadByte();
+
+                        if (b != -1)
+                            decompressorOutput.WriteByte((byte)b);
+                        else
+                            finished = true;
+                    }
+                    else
+                    {
+                        retCount = decompressor.Read(bytes, 0, bufferSize);
+
+                        if (retCount != 0)
+                            decompressorOutput.Write(bytes, 0, retCount);
+                        else
+                            finished = true;
+                    }
+                }
+                decompressor.Dispose();
+                decompressorOutput.Position = 0;
+
+                byte[] decompressorOutputBytes = decompressorOutput.ToArray();
+                byte[] correctOutputBytes = correctDecompressedOutput.ToArray();
+
+                Assert.Equal(correctOutputBytes.Length, decompressorOutputBytes.Length);
+                for (int i = 0; i < correctOutputBytes.Length; i++)
+                {
+                    Assert.Equal(correctOutputBytes[i], decompressorOutputBytes[i]);
+                }
+            }
+        }
```

Comment:
Are there multiple-streams-in-a-stream input files from other utilities we could also consume as a test, to validate we're treating things the same as other tools that allow this?

Reply:
I couldn't find a case where a tool concatenates the streams for you. In most cases the user does a gzip compress on series of files independently where the output is concatenated with the previous streams output.

e.g.
1. gzip file1 > output.gz
2. gzip file2 >> output.gz

Reply:
What about another platform's library support for this?  All I'm suggesting is we have a file containing multiple streams built with a library other than ours.

Reply:
Ok, thanks.  Let's go ahead and get this in, and then we can add additional tests after as is needed.  From what you've said, it sounds like this isn't a big deal.